contrib/internal/httptrace/httptrace.go

```diff
@@ -105,13 +106,18 @@ func urlFromRequest(r *http.Request) string {
 
 // HeaderTagsFromRequest matches req headers to user-defined list of header tags
 // and creates span tags based on the header tag target and the req header value
-func HeaderTagsFromRequest(req *http.Request, lookupHeader func(string) (string, bool)) ddtrace.StartSpanOption {
+func HeaderTagsFromRequest(req *http.Request, headerCfg internal.LockMap) ddtrace.StartSpanOption {
 	return func(cfg *ddtrace.StartSpanConfig) {
-		for h, v := range req.Header {
-			h = strings.ToLower(h)
-			if tag, ok := lookupHeader(h); ok && !strings.HasPrefix(h, "x-datadog-") {
-				cfg.Tags[tag] = strings.TrimSpace(strings.Join(v, ","))
+		var opts []ddtrace.StartSpanOption
+		headerCfg.Iter(func(header, tag string) {
+			if vs, ok := req.Header[header]; ok {
+				opts = append(opts, func(cfg *ddtrace.StartSpanConfig) {
+					cfg.Tags[tag] = strings.TrimSpace(strings.Join(vs, ","))
```

Comment:
Hmm... Do we even need to build the `opts` slice? 

Can't we just set `cfg.Tags` here rather than appending a closure? Then remove lines 119-121?